keyvault/spring-cloud-azure-starter-keyvault-secrets/single-property-source/README.md

```diff
@@ -1,21 +1,135 @@
+- [An Azure subscription](https://azure.microsoft.com/free/)
+- [Terraform](https://www.terraform.io/)
+- [IntelliJ IDEA](https://www.jetbrains.com/idea/download/#section=mac)
+- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- JDK8
+- Maven
+
+## Prevision Azure Resources required to run this sample.
+
+### Authenticate using the Azure CLI
+Terraform must authenticate to Azure to create infrastructure.
+
+In your terminal, use the Azure CLI tool to setup your account permissions locally.
+
+```shell
+az login
+```
+
+Your browser window will open and you will be prompted to enter your Azure login credentials. After successful authentication, your terminal will display your subscription information. You do not need to save this output as it is saved in your system for Terraform to use.
+
+```shell
+You have logged in. Now let us find all the subscriptions to which you have access...
+
+[
+  {
+    "cloudName": "AzureCloud",
+    "homeTenantId": "home-Tenant-Id",
+    "id": "subscription-id",
+    "isDefault": true,
+    "managedByTenants": [],
+    "name": "Subscription-Name",
+    "state": "Enabled",
+    "tenantId": "0envbwi39-TenantId",
+    "user": {
+      "name": "your-username@domain.com",
+      "type": "user"
+    }
+  }
+]
+```
+
+If you have more than one subscription, specify the subscription-id you want to use with command below:
+```shell
+az account set --subscription <your-subscription-id>
+```
+
+### Provision the Resources
+
+After login Azure CLI with your account, now you can use the terraform script to create Azure Resources.
+
+```shell
+# In the root directory of the sample
+# Initialize your Terraform configuration
+terraform -chdir=./terraform init
+
+# Apply your Terraform Configuration
+# Type `yes` at the confirmation prompt to proceed.
+terraform -chdir=./terraform apply
+
+```
+
+
+It may take a few minutes to run the script. After successful running, you will see prompt information like below:
+
+```shell
+
+azurecaf_name.kv: Creating...
+azurecaf_name.resource_group: Creating...
+azurecaf_name.resource_group: Creation complete after ...
+azurecaf_name.kv: Creation complete after 0s ...
+azurerm_resource_group.main: Creating...
+azurerm_resource_group.main: Creation complete after 4s ...
+azurerm_key_vault.kv_account: Creating...
+azurerm_key_vault.kv_account: Still creating... 
+...
+azurerm_key_vault.kv_account: Creation complete after 2m36s ...
+azurerm_key_vault_secret.kv: Creating...
+azurerm_key_vault_secret.kv: Creation complete after 8s ...
+
+Apply complete! Resources: 5 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+...
+
+
+```
+
+You can go to [Azure portal](https://ms.portal.azure.com/) in your web browser to check the resources you created.
+
+### Export output to your local Environment
+Running the command below to export environment values:
+
+```shell
+ source ./terraform/setup_env.sh
+```
+
+## Run locally
```

Comment:
Should we change to Run Locally here?

Reply:
maybe, we will add something like # run in asc, that's why ;
for now, it's ok to remove it.